app/controllers/concerns/current_stage.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+module CurrentStage
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :require_stage
+    helper_method :current_stage
+  end
+
+  def current_stage
+    @stage
+  end
+
+  protected
+
+  def require_stage
+    @stage = (Stage.find_by_param!(params[:stage_id]) if params[:stage_id])
```

Comment:
this won't work since stage permalinks are not globally unique, need to find base on project